src/js/date.js

```diff
@@ -1,6 +1,9 @@
 export const DEFAULT_NEW_YEARS = 10;
 export const TODAY = 'TODAY';
 
+// https://en.wikipedia.org/wiki/Date_format_by_country
+const ALL_DATE_SEPERATORS = [' ', '.', '/', '-'];
```

Comment:
Maybe better to use:
https://github.com/Brightspace/intl

> Why not just use the standard [ECMAScript Internationalization API (ECMA-402)](http://www.ecma-international.org/ecma-402/1.0/#sec-8) and [related polyfills](https://github.com/andyearnshaw/Intl.js)? Firstly, **the standard doesn't include any parsing functionality**. Additionally, Brightspace supports fine-grained locale customization at the organization and user levels -- a level of configuration that simply isn't present in the standard. This library does attempt to follow the standard API syntax and naming conventions when possible.

Reply:
IMHO there are dozens of various locale files and we should rely on proper parsing based on concrete rules.

Reply:
not necceary to add another dependecy when those 4 seperators are internationally defined

Reply:
but thanks

Reply:
https://en.wikipedia.org/wiki/Date_format_by_country

Reply:
Please be carefull and watch out.
Those separates do not tell you how date information is order - i.e big-endian, little-endian, middle-endian.
And for proper parsing we need this information of `YYYY-MM-DD` vs `MM-DD-YYYY` vs `DD-YYYY-MM` and all other combination used in different locales.

Reply:
that part is covered below in the logic

Reply:
that is what the blueprint is for

Reply:
here we ONLY want to know on what to split (`-`, `.`, etc). Below we get the position for a certain locale using the blueprint

Reply:
it also includes YY or YYYY and MM or M.

Reply:
Yeah I got that.
It's clever though this way we have to trust a browsers implementation which maybe buggy.
And I know to do it ourselves is way to much work and very detailed...

Reply:
agree in not re-inventing the wheel, but am sorry if works in all our browsers we need to support with YYYY-DD-MM and YYYY-MM-DD and MM-DD-YY and so on, why do we have to use a library? I mean why investing time and add a dependecy to another npm module if there is not evidence yet that native browsers have issue in implementing a standard api? I get your concern, but i think here is maybe a bit of overthinking. or am i wrong?

Reply:
every npm dependecy decreases maintability and compatibility to other frontend stacks